### DISCUSSION TITLE
npm run build is returning exit code 0 to CircleCI

### DISCUSSION THREAD

**Discussion by jasonhilldm** [preactjs/preact-cli#1743] [Q&A] [answered]
I am using CircleCI to deploy an app when changes are pushed to git. One of the steps is the command "npm run build" which in turn runs "preact build".

When there is an error with the build, I am seeing a "Build failed!" message in the CircleCI step but it also says "CircleCI received exit code 0" and so continues with the next step of the deployment.

Any ideas why CircleCI is receiving an exit code 0 from the build step even though it has failed?

I see a very old issue about this same problem but it was fixed: https://github.com/preactjs/preact-cli/issues/61

**Comment by rschristian** [comment, marked as answer]
~~Do you have an example of an error? It's possible we're missing the correct exit code somewhere, but I'd need a pointer to help track that down.~~

Edit: Ah, I think I spot it. I thought "Build failed!" was our generic message prepended to all errors. It's not.

Opened #1744 for tracking

**Reply by jasonhilldm** [reply]
OK...here's a screenshot of the build step anyway.

![image](https://user-images.githubusercontent.com/87751637/193488212-ddeb0320-3fb6-4142-a668-f294d3869c1b.png)

**Reply by rschristian** [reply]
~~Sorry for the long-time no-fix, but any chance you know how to reproduce this? Was taking a look this evening and I can't figure out how in the world you managed to trigger it.~~

https://github.com/preactjs/preact-cli/blob/12e230d45988ead50a4359b3b8a0228ea6c06e9d/packages/cli/src/lib/webpack/run-webpack.js#L80-L102

~~This is our production compiler code; in short, upon compilation, it'll show errors & warnings to the user (`showStats`). Now, if there are errors, this will be passed to `error()` that outputs a formatted error message and exits with the correct status code (`isProd ? 1 : 0`).~~

~~Somehow, your build produced either no stats or the stats had no errors, yet the Webpack compilation did error; this allowed it to get past `showStats()` and hit that promise rejection instead, which your screenshot shows.~~

~~I have absolutely no clue how this happened tbh. That promise rejection should be dead / impossible to hit code. I tried reverse engineering your provided error message (adding a dummy `import Test from './routes/Test'` in `src/index.js`) but that exits correctly.~~

Edit: Ah, remembered that this can be done if you've made some invalid alterations to the Webpack config. That'll do it. Not dead code after all.

**Reply by rschristian** [reply]
Resolved in #1754
